ilc/server/registry/Registry.js

```diff
@@ -50,13 +50,18 @@ module.exports = class Registry {
         this.#logger.info('Registry preheated successfully!');
     }
 
-    #getConfig = async () => {
+    #getConfig = async (options) => {
         this.#logger.debug('Calling get config registry endpoint...');
 
         const tplUrl = urljoin(this.#address, 'api/v1/config');
         let res;
         try {
             res = await axios.get(tplUrl, { responseType: 'json' });
+
+            const { domain } = options?.filter || {};
+            if (domain) {
+                res.data = this.#filterConfigByDomain(res.data, domain);
```

Comment:
Pass `options?.filter` directly to `this.#filterConfigByDomain` & rename it to `this.#filterConfig`

Reply:
done